zproject/prod_settings_template.py

```diff
@@ -166,21 +166,20 @@
 # directly on the Zulip server.  If file storage in Amazon S3 is
 # desired, you can configure that as follows:
 #
-# (1) Set s3_key and s3_secret_key in /etc/zulip/zulip-secrets.conf to
+# Set s3_key and s3_secret_key in /etc/zulip/zulip-secrets.conf to
 # be the S3 access and secret keys that you want to use, and setting
 # the S3_AUTH_UPLOADS_BUCKET and S3_AVATAR_BUCKET to be the S3 buckets
 # you've created to store file uploads and user avatars, respectively.
 # Then restart Zulip (scripts/restart-zulip).
-#
-# (2) Edit /etc/nginx/sites-available/zulip-enterprise to comment out
-# the nginx configuration for /user_uploads and /user_avatars (see
-# https://github.com/zulip/zulip/issues/291 for discussion of a better
-# solution that won't be automatically reverted by the Zulip upgrade
-# script), and then restart nginx.
 LOCAL_UPLOADS_DIR = "/home/zulip/uploads"
 #S3_AUTH_UPLOADS_BUCKET = ""
 #S3_AVATAR_BUCKET = ""
 
+# Setup the version of NGINX which is being used for this installation.
+# By default a NGINX version greater that 1.5.9 is assumed but if you are using
+# a older version, it is necessary to specify it here.
+# NGINX_VERSION = '1.4.6'
```

Comment:
While django-sendfile doesn't want to auto-detect, I think we should auto-detect, because which `nginx` version is being used is mostly a function of whether it's Trusty or Xenial, and the software should know that.  It's really fast to check if the server has `nginx` installed.  

```
time nginx -v
nginx version: nginx/1.4.6 (Ubuntu)

real	0m0.009s
user	0m0.009s
sys	0m0.000s
```

What I'd do is something like this block in `zproject/settings.py`:

```
if settings.PRODUCTION and settings.LOCAL_UPLOADS_DIR:
    detected_nginx_version = subprocess.check_output(["nginx", "-v"]) # Needs further string processing to be correct.
else:
    detected_nginx_version = None
DEFAULT_SETTINGS["NGINX_VERSION"] = detected_nginx_version
```